simulator/func_sim/alu.h

```diff
@@ -285,6 +286,13 @@ struct ALU
         instr->v_dst[0] = ( (instr->v_src[0] >> pack_width) | (instr->v_src[1] & (bitmask<T>(pack_width) << pack_width)));
     }
 
+    static void bclri( Instr* instr )
+    {
+        size_t index = shamt_imm( instr) & (XLEN - 1);
+        RegisterUInt mask = ~(static_cast<RegisterUInt>( 1) << index);
```

Comment:
Let's use `auto` for types. Also, use `lsb_set<RegisterUInt>` to get `1` constant of proper size.